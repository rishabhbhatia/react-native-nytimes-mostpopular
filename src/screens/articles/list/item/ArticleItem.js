import React from "react";
import { TouchableWithoutFeedback, View, Text } from "react-native";

import Icon from "react-native-vector-icons/FontAwesome";

import styles from "./styles";

const ArticleItem = (props) => {
  const { article = {}, onPress } = props;
  const { title = "", byline = "", published_date: publishedDate = "" } = article;

  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={styles.row}>
        <View style={styles.itemIndicatorContainer}>
          <View style={styles.circle} />
        </View>
        <View style={styles.descriptionContainer}>
          <Text style={styles.title} numberOfLines={2} ellipsizeMode="tail">
            {title}
          </Text>
          <Text style={styles.byline}>{byline}</Text>
          <View style={styles.dateContainer}>
            <Icon name="calendar" size={10} color="#969696" />
            <Text style={styles.publishedDate}>{publishedDate}</Text>
          </View>
        </View>
        <View style={styles.detailIndicatorContainer}>
          <Icon name="chevron-right" size={15} color="#c9c7c7" />
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default ArticleItem;
