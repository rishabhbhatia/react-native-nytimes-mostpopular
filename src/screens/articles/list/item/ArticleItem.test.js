import React from "react";
import { shallow } from "enzyme";
import ArticleItem from "./ArticleItem.js";

/* -------------------------------------------------------------------------- */
/*                                  UI Tests                                  */
/* -------------------------------------------------------------------------- */
describe("ArticleItem", () => {
    /* -------------------------------- snapshot -------------------------------- */
    it("matches snapshot", () => {
        expect.assertions(1);
        const articleItem = shallow(<ArticleItem />);
        expect(articleItem).toMatchSnapshot();
    });
});
