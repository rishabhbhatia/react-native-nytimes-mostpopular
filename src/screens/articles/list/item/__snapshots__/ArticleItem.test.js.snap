// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem matches snapshot 1`] = `
<TouchableWithoutFeedback>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#969696",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 0.6,
          "justifyContent": "center",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "500",
            "textAlign": "left",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#9e9e9e",
            "fontSize": 12,
            "fontWeight": "400",
            "marginTop": 10,
            "textAlign": "left",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginTop": 2,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#969696"
          name="calendar"
          size={10}
        />
        <Text
          style={
            Object {
              "color": "#9e9e9e",
              "fontSize": 10,
              "fontWeight": "400",
              "marginLeft": 5,
              "textAlign": "right",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "center",
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#c9c7c7"
        name="chevron-right"
        size={15}
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;
